rinja_derive/src/generator.rs

```diff
@@ -1962,28 +1919,59 @@ struct Buffer {
     // The buffer to generate the code into
     buf: String,
     discard: bool,
+    last_was_write_str: bool,
 }
 
 impl Buffer {
     fn new() -> Self {
         Self {
             buf: String::new(),
             discard: false,
+            last_was_write_str: false,
         }
     }
 
+    fn is_discard(&self) -> bool {
+        self.discard
+    }
+
+    fn set_discard(&mut self, discard: bool) {
+        self.discard = discard;
+        self.last_was_write_str = false;
+    }
+
     fn writeln(&mut self, src: impl BufferFmt) {
         if !self.discard {
             src.append_to(&mut self.buf);
             self.buf.push('\n');
+            self.last_was_write_str = false;
         }
     }
 
     fn write(&mut self, src: impl BufferFmt) {
         if !self.discard {
             src.append_to(&mut self.buf);
+            self.last_was_write_str = false;
         }
     }
+
+    fn write_writer(&mut self, s: &str) -> usize {
+        if self.discard {
+            // nothing to do
+        } else if !self.last_was_write_str {
+            writeln!(self.buf, "writer.write_str({s:#?})?;").unwrap();
+            self.last_was_write_str = true;
+        } else {
+            // strip trailing `")?\n`, leaving an unterminated string
+            let len = self.buf.strip_suffix("\")?;\n").unwrap().len();
```

Comment:
I'm worried about whether or not this `strip_suffix` will work on windows since it's `\r\n` and we use `writeln` a lot. Maybe we should stop printing backlines? It's not like they're really useful...

Reply:
I removed any newlines generation. Until #87 the generated code cannot be compared anymore. #87 lets `syn` do the comparison, and `syn` does not care about spacing.

Reply:
Can be rebased now. ;)